src/main/kotlin/anttracker/PageOf.kt

```diff
@@ -0,0 +1,166 @@
+    // Getter for elements in the records MutableList.
+    // ---
+    fun getContentAt(index: Int): IntEntity = records[index]
+
+    // -------------------------------------------------------------------------------
+    // Increments page number and loads associated DB records into records MutableList
+    // Throws an error if attempting to load beyond the last page.
+    // ---
+    fun loadNextPage() {
+        if (lastPage()) {
+            throw Exception("PageOf: Already reached last page.")
+        }
+        pagenum++
+        loadRecords()
+    }
+
+    // -------------------------------------------------------------------------------
+    // TRUE if current page >= than the last page.
+    // ---
+    fun lastPage(): Boolean = pagenum >= lastPageNum
+
+    // -------------------------------------------------------------------------------
+    // Returns the number of yet-to-be-displayed records after the current page.
+    // ---
+    fun countRemainingRecords(): Int = getQueryRecordCount() - (qLimit * (pagenum + 1))
+
+    // -------------------------------------------------------------------------------
+    // Used in init/ctor block to calculate the last page number.
+    // As we are 0-indexing the page numbers, we subtract 1 at the end.
+    // ---
+    private fun initLastPageNum() {
+        lastPageNum =
+            ceil(
+                (getQueryRecordCount().toDouble() / qLimit.toDouble()),
+            ).toInt() - 1
+    }
+
+    // -------------------------------------------------------------------------------
+    // Returns the total number of records/rows of the query being paginated.
+    // Used for calculating the number of pages the query needs.
+    // ---
+    private fun getQueryRecordCount(): Int {
+        val numRecords: Int =
+            queryToDB()?.count()?.toInt()
+                ?: throw IllegalArgumentException("Error: Query returned null")
+        return numRecords
+    }
+
+    // -------------------------------------------------------------------------------
+    // Defines the DAO query to DB used to pull records into memory.
+    // Abstract/Virtual as query needs to be defined per PageOf Type
+    // ---
+    protected abstract fun queryToDB(): SizedIterable<IntEntity>?
+
+    // -------------------------------------------------------------------------------
+    // Defines how a single record is printed to console.
+    // Abstracted as this changes based on record type (Issue, Contact, Release, etc.)
+    // ---
+    protected abstract fun printRecord(record: IntEntity)
+}
+
+// -------------------------------------------------------------------------------
+// Implementation of a PageOf Class as PageOfReleases
+// Each PageOf class needs to define:
+//      - display(), to define how the page is displayed to console
+//      - queryToDB(), to define the DAO query to DB used to pull records into memory
+// ---
+class PageOfReleases(
```

Comment:
Thoughts on moving this to Release.kt? Or will this be used in other modules?

Reply:
I don't think there'd be any issues moving PageOf subclasses to their respective modules; it'd be in line with our roundabout design. Will do.